dataframely/collection.py

```diff
@@ -561,6 +561,42 @@ def filter(
 
         return cls._init(results), failures
 
+    def join(
+        self,
+        primary_keys: pl.LazyFrame,
+        how: Literal["semi", "anti"] = "semi",
+    ) -> Self:
+        """Filter the collection by joining onto a data frame containing entries for the
```

Comment:
I'm wondering if we still need an `Attention:` block that says that it's the user's responsibility that the resulting collection is still valid (just like in `cast`). You can probably construct custom rules that could be violated by running this join.